src/Config.hs

```diff
@@ -0,0 +1,223 @@
+    token = strOption
+      $  long "token"
+      <> metavar "TOKEN"
+      <> readValueFromEnv "VAULT_TOKEN" envVars
+      <> help "Token to authenticate to Vault with. Also configurable via VAULT_TOKEN."
+    secretsFile = strOption
+      $  long "secrets-file"
+      <> metavar "FILENAME"
+      <> readValueFromEnv "VAULTENV_SECRETS_FILE" envVars
+      <> help ("Config file specifying which secrets to request. Also configurable " ++
+               "via VAULTENV_SECRETS_FILE." )
+    cmd = argument str
+      $  metavar "CMD"
+      <> help "command to run after fetching secrets"
+    cmdArgs =  many $ argument str
+      (  metavar "ARGS..."
+      <> help "Arguments to pass to CMD, defaults to nothing")
+    noConnectTls =
+       flag (efNoConnectTls envFlags) True
+        (  long "no-connect-tls"
+        <> help ("Don't use TLS when connecting to Vault. Default: use TLS. Also " ++
+                "configurable via VAULTENV_NO_CONNECT_TLS."))
+      <|> flag (efNoConnectTls envFlags) False
+        (  long "connect-tls"
+        <> help ("Always connect to Vault via TLS. Default: use TLS. Can be used " ++
+                 "to override VAULTENV_NO_CONNECT_TLS."))
+    noValidateCerts =
+      flag (efNoValidateCerts envFlags) True
+        (  long "no-validate-certs"
+        <> help ("Don't validate TLS certificates when connecting to Vault. Default: " ++
+                "validate certs. Also configurable via VAULTENV_NO_VALIDATE_CERTS."))
+      <|> flag (efNoValidateCerts envFlags) False
+        (  long "validate-certs"
+        <> help ("Always validate TLS certificates when connecting to Vault. Default: " ++
+                 "validate certs. Can be used to override VAULTENV_NO_CONNECT_TLS."))
+    noInheritEnv =
+      flag (efNoInheritEnv envFlags) True
+        (  long "no-inherit-env"
+        <> help ("Don't merge the parent environment with the secrets file. Default: " ++
+                "merge environments. Also configurable via VAULTENV_NO_INHERIT_ENV."))
+      <|> flag (efNoInheritEnv envFlags) False
+        (  long "inherit-env"
+        <> help ("Always merge the parent environment with the secrets file. Default: " ++
+                 "merge environments. Can be used to override VAULTENV_NO_INHERIT_ENV.")
+        )
+    baseDelayMs = MilliSeconds <$> (option auto
+      $  long "retry-base-delay-milliseconds"
+      <> metavar "MILLISECONDS"
+      <> readValueFromEnvWithDefault "VAULTENV_RETRY_BASE_DELAY_MS" 40 envVars
+      <> help ("Base delay for vault connection retrying. Defaults to 40ms. " ++
+                "Also configurable via VAULTENV_RETRY_BASE_DELAY_MS."))
+    retryAttempts = option auto
+      $  long "retry-attempts"
+      <> metavar "NUM"
+      <> readValueFromEnvWithDefault "VAULTENV_RETRY_ATTEMPTS" 9 envVars
+      <> help "Maximum number of vault connection retries. Defaults to 9"
+
+-- ! Attempt to parse an optparse default value modifier from a list of
```

Comment:
s/-- !/-- |  (everywhere in this module)

Reply:
Fixed